### PR TITLE
Migrate the Windows build to build.ps1, a new build script

### PR DESCRIPTION
**Pull request by tristanlabelle**
Migrates the Windows CI build script to [`build.ps1`](https://github.com/compnerd/swift-build/blob/main/build.ps1), a PowerShell-based build script suitable for local incremental builds as well as CI builds, including multi-architecture builds, packaging and test running.

### CODE REVIEW

**Comment by compnerd** [comment]
@swift-ci please build toolchain Windows platform

**Comment by compnerd** [comment]
@swift-ci please build toolchain Windows platform

**Comment by compnerd** [comment]
@swift-ci please build toolchain Windows platform

**Comment by compnerd** [comment]
@swift-ci please build toolchain windows platform

File: utils/build-windows-toolchain.bat [hunk 1]
```diff
@@ -824,193 +129,7 @@ git clone --quiet --no-tags --depth 1 --branch v1.2.11 https://github.com/madler
...
-md %BuildRoot%\artifacts\offline
-
-move %PackageRoot%\offline\Release\amd64\installer.exe %BuildRoot%\artifacts\offline || (exit /b)
-move %PackageRoot%\online\Release\amd64\*.cab %BuildRoot%\artifacts\online\ || (exit /b)
-move %PackageRoot%\online\Release\amd64\*.exe %BuildRoot%\artifacts\online\ || (exit /b)
-move %PackageRoot%\online\Release\amd64\*.msi %BuildRoot%\artifacts\online\ || (exit /b)
-
-:: Workaround for lack of control over Jenkins ...
-copy %BuildRoot%\artifacts\offline\installer.exe %BuildRoot%\artifacts\
+git clone --quiet --no-tags --depth 1 --branch curl-7_77_0 https://github.com/curl/curl
```
Comment (shahmishal) [inline utils/build-windows-toolchain.bat:132]: This needs to be part of update-checkout, we should not clone repos outside of update-checkout.

Comment (shahmishal) [reply utils/build-windows-toolchain.bat:132]: Also, this version is wrong it should 8.4.x

File: utils/build-windows-toolchain.bat [hunk 2]
```diff
@@ -824,193 +129,7 @@ git clone --quiet --no-tags --depth 1 --branch v1.2.11 https://github.com/madler
 git clone --quiet --no-tags --depth 1 --branch v2.9.12 https://github.com/gnome/libxml2
 git clone --quiet --no-tags --depth 1 --branch version-3.36.0 https://github.com/sqlite/sqlite
 git clone --quiet --no-tags --depth 1 --branch maint/maint-69 https://github.com/unicode-org/icu
```
Comment (shahmishal) [inline utils/build-windows-toolchain.bat:131]: We should also move these to update-checkout.

**Comment by compnerd** [comment]
@swift-ci please build toolchain Windows platform

File: utils/build-windows-toolchain.bat (hunk 2 above)
Comment (stevapple) [reply utils/build-windows-toolchain.bat]: https://github.com/apple/swift/issues/60457

**Comment by compnerd** [comment]
@swift-ci please smoke test

**Comment by shahmishal** [comment]
The new script is missing `.msi` files, only see these two files. is that expected?

- installer.exe 
- rtl.amd64.msm

**Comment by stevapple** [comment]
> The new script is missing `.msi` files, only see these two files. is that expected?
> 
> * installer.exe
> * rtl.amd64.msm

Seems just not copied? https://github.com/tristanlabelle/swift/blob/20c89b5d792f2ab21819b0f0792607057ba398dd/utils/build.ps1#L1581-L1582

**Comment by compnerd** [comment]
@shahmishal yeah, that is intentional - the artifacts were what we wanted to actually have generally available to download. The msm is the redistributable merge module that allows others to integrate a distribution of the runtime (due to lack of ABI stability). The installer is the installer for the toolchain, which now allows you to control what is installed.

I can certainly add in the MSIs if you like.

**Comment by compnerd** [comment]
@swift-ci please build toolchain Windows platform

**Comment by compnerd** [comment]
@swift-ci please smoke test

**Comment by compnerd** [comment]
@swift-ci please build toolchain Windows platform

**Comment by compnerd** [comment]
@swift-ci please build toolchain Windows platform

**Comment by shahmishal** [comment]
@swift-ci please build toolchain Windows platform

**Comment by compnerd** [comment]
@swift-ci please smoke test

**Pull request merged**